### PR TITLE
Refactor daa signal

### PR DESCRIPTION
**Pull request by tspenov**
Discord allows max 2000 chars per message. Split into smaller messages.
Refactor using the behaviour.

### CODE REVIEW

File: lib/sanbase/notifications/discord/exchange_inflow.ex [hunk 1]
```diff
@@ -19,7 +19,10 @@ defmodule Sanbase.Notifications.Discord.ExchangeInflow do
   def run() do
     projects =
       projects()
-      |> filter_projects()
+      |> Project.projects_over_volume_threshold(
+        Config.get(:trading_volume_threshold)
+        |> String.to_integer()
+      )
```
Comment (IvanIvanoff) [inline lib/sanbase/notifications/discord/exchange_inflow.ex]: Can you put this in a separate variable defined before `projects = ... `

File: lib/sanbase/notifications/discord/discord.ex [hunk 2]
```diff
@@ -5,17 +5,40 @@ defmodule Sanbase.Notifications.Discord do
...
     case http_client().post(webhook, payload, [{"Content-Type", "application/json"}]) do
       {:ok, %HTTPoison.Response{status_code: code}} when code in 200..299 ->
         :ok
 
+      {:ok, %HTTPoison.Response{status_code: 429, body: body} = resp} ->
+        body = body |> Jason.decode!()
+
+        Logger.warn(
+          "Cannot publish #{signal_name} signal in Discord: HTTP Response: #{inspect(resp)}"
+        )
```
Comment (IvanIvanoff) [inline lib/sanbase/notifications/discord/discord.ex]: This case is expected so it probably should not be logged at a `warn` level

File: lib/sanbase/notifications/discord/discord.ex [hunk 3]
```diff
@@ -5,17 +5,40 @@ defmodule Sanbase.Notifications.Discord do
...
+
+        Process.sleep(body["retry_after"] + 1000)
+
+        send_notification(webhook, signal_name, payload, %{
+          opts
+          | retry_count: Map.get(opts, :retry_count, 0) + 1
+        })
```
Comment (IvanIvanoff) [inline lib/sanbase/notifications/discord/discord.ex]: You can simplify this with:
```elixir
send_notification(webhook, signal_name, payload, Map.update(opts, :retry_count, 0, & &1+1))

File: lib/sanbase/notifications/discord/discord.ex (hunk 2 above)
Comment (tspenov) [reply lib/sanbase/notifications/discord/discord.ex]: lets have at least info

**Pull request merged**